internal/storage/postgresql.go

```diff
@@ -0,0 +1,99 @@
+package storage
+
+import (
+	"github.com/jinzhu/gorm"
+	_ "github.com/jinzhu/gorm/dialects/postgres"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// implements Storage interface
+type PostgresqlStorage struct {
+	connectionString string
+}
+
+func (s *PostgresqlStorage) conn() (*gorm.DB, error) {
+	db, err := gorm.Open("postgres", s.connectionString)
+	if err != nil {
+		logrus.Fatal(errors.Wrap(err, "failed to connect to postgresql"))
+	}
+	// db.LogMode(true)
+	return db, err
+}
+
+func NewPostgresStorage(connectionString string) *PostgresqlStorage {
+	s := &PostgresqlStorage{connectionString}
+	db, err := s.conn()
+	defer db.Close()
+	if err != nil {
+		panic(err)
+	}
+
+	// Migrate the schema
+	db.AutoMigrate(&Device{})
+	return s
+}
+
+func (s *PostgresqlStorage) Save(device *Device) error {
+	db, err := s.conn()
```

Comment:
does that work? when i used normal sql in the past, it would just eat up all connections and then die. Is that what you meant by connection pool? I can totally redo it without the defer everywhere, its easier and cleaner, i was just paranoid about leaking